### PR TITLE
XML Generator for Elementor Widgets

### PR DESCRIPTION
**Pull request by diiegopereira**
See compsupp-6861

### CODE REVIEW

**Review by strategio — APPROVED** [review]
@diiegopereira, looks good to me! It will help a lot +1

I am just leaving some comments about formatting (if you want to upgrade the code):
- Now we just use shorthand arrays (`[]` instead of `array()`)
- We give some spaces inside parenthesis or brackets with variable keys.
- We try to align equal signs.

You can to follow the WP PHP coding standards => https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/

Let me know if you want me to merge right now or not.

**Comment by diiegopereira** [comment]
Thanks @strategio ! I improved the code formatting. If it's ok, we can merge.

File: inc/class-mltools-elementor-config-generator.php [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+				$widgetType                    = $element->widgetType;
+				$settings                      = (array) get_object_vars( $element->settings );
+		
+				foreach ( $settings as $field_key => $field_value ) {
+					$settings[ $field_key ]   = $this->get_field_from_widget( $field_key, $field_value, $widgetType );
+				}
+		
+				$widgets[ $widgetType ]       = [];
+				$widgets[ $widgetType ]['widgetType'] = $widgetType;
+				$widgets[ $widgetType ]['settings']   = $settings;
```
Comment (dgwatkins) [inline inc/class-mltools-elementor-config-generator.php]: @diiegopereira I am not sure how far you want to go with this.

The 3 equal signs above should be aligned according to standards.

File: inc/class-mltools-elementor-config-generator.php [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	public function get_widgets_list( $elements ) {
+		$widgets = [];
+		
+		foreach ( $elements as $element ) {
+			if ( $element->elType === 'widget' ) {
+				$widgetType                    = $element->widgetType;
+				$settings                      = (array) get_object_vars( $element->settings );
+		
+				foreach ( $settings as $field_key => $field_value ) {
+					$settings[ $field_key ]   = $this->get_field_from_widget( $field_key, $field_value, $widgetType );
```
Comment (dgwatkins) [inline inc/class-mltools-elementor-config-generator.php]: This one has too many spaces @diiegopereira 
```suggestion
					$settings[ $field_key ] = $this->get_field_from_widget( $field_key, $field_value, $widgetType );
```

File: inc/class-mltools-elementor-config-generator.php [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+				}
+		
+				$widgets[ $widgetType ]       = [];
+				$widgets[ $widgetType ]['widgetType'] = $widgetType;
+				$widgets[ $widgetType ]['settings']   = $settings;
+			}
+			if ( ! empty( $element->elements ) ) {
+				$widgets                      = array_merge( $widgets, $this->get_widgets_list( $element->elements ) );
```
Comment (dgwatkins) [inline inc/class-mltools-elementor-config-generator.php]: @diiegopereira and this one too:
```suggestion
				$widgets = array_merge( $widgets, $this->get_widgets_list( $element->elements ) );
```

File: inc/class-mltools-elementor-config-generator.php [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+		return $blacklist;
+	}
+	
+	public function get_widgets_list( $elements ) {
+		$widgets = [];
+		
+		foreach ( $elements as $element ) {
+			if ( $element->elType === 'widget' ) {
+				$widgetType                    = $element->widgetType;
+				$settings                      = (array) get_object_vars( $element->settings );
```
Comment (dgwatkins) [inline inc/class-mltools-elementor-config-generator.php]: ... and this one:
```suggestion
				$widgetType = $element->widgetType;
				$settings   = (array) get_object_vars( $element->settings );
```

**Review by dgwatkins — CHANGES_REQUESTED** [review]
I'll take over @diiegopereira because Pierre wont be around until next Tuesday.

One of the coding standard says that equal signs need to be aligned. And with the least number of spaces.

I made a few comments but there are more instances of this mistake. 

Since you are bringing the code up to standards, I thought this would be a good update.

**Comment by diiegopereira** [comment]
@dgwatkins sorry, I tried to format manually + with ChatGPT but it not worked as expected.

Now, Instead of doing manually, I tried to format the code using php-cs-fixer and wp-php-cs-fixer (so it can be done automatically in the future). Could you please check if it worked correctly?

**Review by dgwatkins — APPROVED** [review]
It looks so much better now.

**Pull request merged**